src/libraries/System.Security.Cryptography/tests/X509Certificates/DynamicChainTests.cs

```diff
@@ -594,6 +597,107 @@ public static void NameConstraintViolation_InvalidGeneralNames()
             });
         }
 
+        [ConditionalFact]
+        public static void NameConstraintViolation_ExcludedTree_Upn()
+        {
+            if (PlatformDetection.UsesAppleCrypto && !AppleHasExcludedSubTreeHandling)
+            {
+                throw new SkipTestException("Platform does not handle excludedSubtrees correctly.");
+            }
+
+            SubjectAlternativeNameBuilder builder = new SubjectAlternativeNameBuilder();
+            builder.AddUserPrincipalName("v@example.com");
+
+            AsnWriter writer = new(AsnEncodingRules.DER);
+            writer.WriteCharacterString(UniversalTagNumber.UTF8String, "@example.com");
+            byte[] exampleCom = writer.Encode();
+            writer.Reset();
+
+            NameConstraintsAsn nameConstraints = new NameConstraintsAsn
+            {
+                ExcludedSubtrees =
+                [
+                    new GeneralSubtreeAsn
+                    {
+                        Base = new GeneralNameAsn
+                        {
+                            OtherName = new OtherNameAsn
+                            {
+                                TypeId = "1.3.6.1.4.1.311.20.2.3", //User Principal Name (UPN)
+                                Value = exampleCom,
+                            }
+                        }
+                    }
+                ]
+            };
+
+            nameConstraints.Encode(writer);
+            string encoded = writer.Encode(Convert.ToHexString);
+
+            TestNameConstrainedChain(encoded, builder, (bool result, X509Chain chain) => {
+                Assert.False(result, "chain.Build");
+
+                if (PlatformDetection.IsWindows)
+                {
+                    Assert.Equal(X509ChainStatusFlags.HasExcludedNameConstraint, chain.AllStatusFlags());
+                }
+                else
+                {
+                    Assert.Equal(
+                        PlatformNameConstraints(X509ChainStatusFlags.HasNotSupportedNameConstraint),
```

Comment:
Both added tests are negative.  Should they have variants that expect success on Windows?